### PR TITLE
MGDCTRS-1726 feat: pricing tier additions

### PR DESCRIPTION
**Pull request by gashcrumb**
This change brings the new connector selection list component into the app and applies the pricing tier updates.  There's numerous updates and fixes to the connector selection list component, including selection handling, no search results and error state handling.  This change also significantly improves the cache updating.

Some known things:
1) placeholder text in the tier info popup, this awaits text I think
2) Info pop-up isn't working next to the connector name, there's still some work to be done on the layout of the list item, this will be fixed in a later update as part of that effort
3)  The padding on the right-hand side is to accommodate the "Feedback" thingie

Screenshot from 2022-12-07 14-57-40

### CODE REVIEW

**Comment by gashcrumb** [comment]
There's a selection issue to look at still, best to hold off merging this until it's sorted.

**Comment by gashcrumb** [comment]
Selection works nicely now, and I added back the selectable over behavior, interestingly enough the help popover will not work out of the box when selection is enabled on the DataList.

**Review by indraraj — APPROVED** [review]

**Comment by gashcrumb** [comment]
Adding the do-not-merge label.

It should be the intersection of the filters.  So if you check 'source' and 'free' and are in featured, it should show only featured connectors that have both the 'source' and 'free' labels.

File: src/apis/api.ts [hunk 1]
```diff
@@ -541,8 +541,28 @@ export const fetchConnectorTypes = ({
...
+              .map((s) => `label = ${s.slice(2, s.length)}`)
+              .join(' AND '),
+          ]
+            .filter(Boolean)
+            .join(' AND ')
         : undefined;
+    /*
+      * label searches could probably use the IN operator
+      label && label.length > 0
+        ? `label IN (${label
```
Comment (gashcrumb) [inline src/apis/api.ts:560]: @dhirajsb when you're back, would this be a nicer alternative?  I'm not a fan of my current approach :-)

**Comment by mmuzikar** [comment]
@gashcrumb I noticed 2 thing whilst playing with this PR: 
1. the numbers of connectors in a category update when the pricing tier is changed (nice), but the same doesn't happen for source/sink - would it be possible to add?
2. this one is strange :D if you scroll in the list before it's been fully loaded (or at least that's what I assume) then the items you scroll to won't get loaded until you scroll over them :dizzy_face: just like this:
[Peek 2023-01-05 09-46.webm](https://user-images.githubusercontent.com/46345469/210739498-a2c48432-d58f-4dd1-afff-61132ee642e3.webm)

**Comment by gashcrumb** [comment]
> the numbers of connectors in a category update when the pricing tier is changed (nice), but the same doesn't happen for source/sink - would it be possible to add?

yep, I think so.

> this one is strange :D if you scroll in the list before it's been fully loaded (or at least that's what I assume) then the items you scroll to won't get loaded until you scroll over them dizzy_face just like this

Yes, we're still learning some things about react-virtualized :-)  I think the fetching in the cache component needs some more tightening up.

**Comment by dhirajsb** [comment]
This is a little complicated in SQL, since the where clause has multiple conditions with different values for the same column. It's solvable with other SQL clauses like GROUP BY or INTERSECT. But we need to discuss this in more detail.

**Comment by gashcrumb** [comment]
On the loading I went ahead and created [MGDCTRS-1831](https://issues.redhat.com/browse/MGDCTRS-1831) as this is something we can fix later I think.  And then for the filtering I've gone and created [MGDCTRS-1832](https://issues.redhat.com/browse/MGDCTRS-1832) to cover updating the UI to use an SQL query that will return the results as we expect.  Thankfully @dhirajsb has been looking at this and it sounds like we could have a solution there.

**Comment by gashcrumb** [comment]
Ok, rebased and ready for another look with the aforementioned caveats :-)

File: src/app/components/ConnectorSelectionList/ConnectorSelectionList.stories.tsx [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
 const API_HOST = 'https://dummy.server';
 
+const SelectionViewInner: FC<ConnectorSelectionListInnerProps> = ({
+  renderSelectionList,
+}) => {
+  const [selectedId, onSelect] = useState<string | undefined>(undefined);
+  return renderSelectionList({
+    selectedId,
+    onSelect: (connector) => {
+      console.log('Selected connector: ', connector);
```
Comment (indraraj) [inline src/app/components/ConnectorSelectionList/ConnectorSelectionList.stories.tsx:24]: do we want to leave this console?

**Review by indraraj — APPROVED** [review]
tested locally (leaving with mentioned issues), looks fine 👍

**Pull request merged**